service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageHearingsControllerHearingInPastMidEventTest.java

```diff
@@ -0,0 +1,142 @@
+        LocalDateTime correctEndDate = LocalDateTime.of(2050, 6, 5, 11, 0);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(parseLong(CASE_ID))
+            .data(Map.of("hearingStartDate", LocalDateTime.of(2001, 6, 5, 11, 0),
+                "hearingEndDate", LocalDateTime.of(2001, 6, 5, 11, 0),
+                "hearingStartDateConfirmation", correctStartDate,
+                "hearingEndDateConfirmation", correctEndDate,
+                "hearingDateConfirmation","No"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "hearing-in-past");
+
+        Map<String, Object> responseData = callbackResponse.getData();
+
+        assertThat(responseData.get("hearingStartDate").equals(correctStartDate));
+        assertThat(responseData.get("hearingEndDate").equals(correctEndDate));
+    }
+
+    @Test
+    void shouldPopulateHearingEndDateToCorrectOneWhenIncorrectAndToggledOn() {
+        given(featureToggleService.isAddHearingsInPastEnabled()).willReturn(true);
+
+        LocalDateTime correctEndDate = LocalDateTime.of(2050, 6, 5, 11, 0);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(parseLong(CASE_ID))
+            .data(Map.of("hearingStartDate", LocalDateTime.of(2001, 6, 5, 11, 0),
+                "hearingEndDate", LocalDateTime.of(2001, 6, 5, 11, 0),
+                "hearingEndDateConfirmation", correctEndDate,
+                "hearingDateConfirmation","No"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "hearing-in-past");
+
+        Map<String, Object> responseData = callbackResponse.getData();
+
+        assertThat(responseData.get("hearingStartDate").equals(LocalDateTime.of(2001, 6, 5, 11, 0)));
+        assertThat(responseData.get("hearingEndDate").equals(correctEndDate));
+    }
+
+    @Test
+    void shouldPopulateHearingStartDateToCorrectOneWhenIncorrectAndToggledOn() {
+        given(featureToggleService.isAddHearingsInPastEnabled()).willReturn(true);
+
+        LocalDateTime correctStartDate = LocalDateTime.of(2050, 6, 5, 11, 0);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(parseLong(CASE_ID))
+            .data(Map.of("hearingStartDate", LocalDateTime.of(2001, 6, 5, 11, 0),
+                "hearingEndDate", LocalDateTime.of(2001, 6, 5, 11, 0),
+                "hearingStartDateConfirmation", correctStartDate,
+                "hearingDateConfirmation","No"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "hearing-in-past");
+
+        Map<String, Object> responseData = callbackResponse.getData();
+
+        assertThat(responseData.get("hearingStartDate").equals(correctStartDate));
+        assertThat(responseData.get("hearingEndDate").equals(LocalDateTime.of(2001, 6, 5, 11, 0)));
+    }
```

Comment:
Again not sure we need to test each field in separate tests when you have a test that covers both.

Reply:
talked about adding param test. - will try tomo